apps/blog/models.py

```diff
@@ -0,0 +1,114 @@
+        max_length=255,
+        verbose_name=ugettext('Category name')
+    )
+    parent_category = models.ForeignKey(
+        'self',
+        on_delete=models.PROTECT,
+        related_name='blog_category',
+        null=True,
+        blank=True,
+        verbose_name=ugettext('Parent category')
+    )
+
+    class Meta:
+        verbose_name = ugettext('Category')
+        verbose_name_plural = ugettext('Categories')
+
+    def __str__(self):
+        return self.name
+
+
+class Blog(models.Model):
+
+    class PublishType(models.TextChoices):
+        PUBLISH = 'publish', 'Publish'
+        DRAFT = 'draft', "Draft"
+
+    title = models.TextField(blank=True, verbose_name=ugettext('Blog title'))
+    description = models.TextField(blank=True, verbose_name=ugettext('Blog Description'))
+    image = models.FileField(
+        upload_to='blog/images/', max_length=255, null=True, blank=True, default=None,
+        verbose_name=ugettext('Blog image')
+    )
+    category = models.ForeignKey(
+        'blog.Category', on_delete=models.CASCADE,
+        related_name='blog_categories', verbose_name=ugettext('Blog category')
+    )
+    tags = models.ManyToManyField(
+        'blog.Tag', related_name='blog_tags', blank=True, verbose_name=ugettext('Blog Tags')
+    )
+    published_date = models.DateField(verbose_name=ugettext('Published Date'))
+    publish_type = models.CharField(
+        choices=PublishType.choices, max_length=40, verbose_name=ugettext('Publish Type'), default=PublishType.DRAFT
+    )
+    # SEO fields
+    meta_title = models.CharField(max_length=255, null=True, blank=True, verbose_name=ugettext('Meta title'))
+    meta_keywords = models.CharField(max_length=255, null=True, blank=True, verbose_name=ugettext('Meta Keywords'))
+    meta_description = models.CharField(max_length=255, null=True, blank=True, verbose_name=ugettext('Meta Description'))
+    og_title = models.CharField(max_length=255, null=True, blank=True, verbose_name=ugettext('Open graph title'))
+    og_description = models.CharField(max_length=255, null=True, blank=True, verbose_name=ugettext('Open graph description'))
+    og_image = models.FileField(
+        upload_to='blog/og_image/', max_length=255, null=True, blank=True, default=None,
+    )
+    og_locale = models.CharField(max_length=255, null=True, blank=True, verbose_name=ugettext('Open graph locale'))
+    og_type = models.CharField(max_length=255, null=True, blank=True, verbose_name=ugettext('Open graph type'))
+
+    # Session to check if image changed to generate og image
+    __image_name = None
+
+    class Meta:
+        verbose_name = ugettext('Blog')
+        verbose_name_plural = ugettext('Blogs')
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__image_name = self.image.name
+
+    def __str__(self):
+        return self.title
+
+    def save(self, *args, **kwargs):
+        super(Blog, self).save(*args, **kwargs)
```

Comment:
Let's move this to last.

Reply:
@thenav56  First we have to save blog object, then we have to save blog object, then if recently blog image is changed, then only we have to generate og image. Because we are generating og_image form image, if image not uploaded it will raise error

Reply:
Oh no... that is another issue. Let's move this.